src/main/java/com/google/firebase/database/core/JvmAuthTokenProvider.java

```diff
@@ -35,51 +35,47 @@
   private final ScheduledExecutorService executorService;
   private final FirebaseApp firebaseApp;
 
-  public JvmAuthTokenProvider(FirebaseApp firebaseApp, ScheduledExecutorService executorService) {
+  JvmAuthTokenProvider(FirebaseApp firebaseApp, ScheduledExecutorService executorService) {
     this.executorService = executorService;
     this.firebaseApp = firebaseApp;
   }
 
-  private static String wrapOAuthToken(FirebaseApp firebaseApp, GetTokenResult result) {
-    String oauthToken = result.getToken();
-    if (oauthToken == null) {
+  private static String wrapOAuthToken(FirebaseApp firebaseApp, AccessToken result) {
+    if (result == null) {
       // This shouldn't happen in the actual production SDK, but can happen in tests.
       return null;
-    } else {
-      Map<String, Object> authVariable = firebaseApp.getOptions().getDatabaseAuthVariableOverride();
-      GAuthToken googleAuthToken = new GAuthToken(oauthToken, authVariable);
-      return googleAuthToken.serializeToString();
     }
+
+    Map<String, Object> authVariable = firebaseApp.getOptions().getDatabaseAuthVariableOverride();
+    GAuthToken googleAuthToken = new GAuthToken(result.getTokenValue(), authVariable);
+    return googleAuthToken.serializeToString();
   }
 
   @Override
   public void getToken(boolean forceRefresh, final GetTokenCompletionListener listener) {
-    ImplFirebaseTrampolines.getToken(firebaseApp, forceRefresh)
-        .addOnCompleteListener(
-            this.executorService,
-            new OnCompleteListener<GetTokenResult>() {
-              @Override
-              public void onComplete(@NonNull Task<GetTokenResult> task) {
-                if (task.isSuccessful()) {
-                  listener.onSuccess(wrapOAuthToken(firebaseApp, task.getResult()));
-                } else {
-                  listener.onError(task.getException().toString());
-                }
-              }
-            });
+    GoogleCredentials credentials = ImplFirebaseTrampolines.getCredentials(firebaseApp);
```

Comment:
Like my feedback elsewhere, it's tempting to cache the credentials object as a private member rather than read it every time (to make it more obvious that it won't change from one request to the next).

Reply:
Done